src/Core/UnifyState.idr

```diff
@@ -576,12 +576,12 @@ checkDelayedHoles
 -- not guarded by a unification problem (in which case, report that the unification
 -- problem is unsolved) and it doesn't depend on an implicit pattern variable
 -- (in which case, perhaps suggest binding it explicitly)
-export
 checkValidHole : {auto c : Ref Ctxt Defs} ->
                  {auto u : Ref UST UState} ->
-                 (Int, (FC, Name)) -> Core ()
-checkValidHole (idx, (fc, n))
-    = do defs <- get Ctxt
+                 Int -> (Int, (FC, Name)) -> Core ()
+checkValidHole base (idx, (fc, n))
+  = if base > idx then pure () else
```

Comment:
We have `unless` btw

Reply:
Oh do we? You'd think I'd know these things. Also I don't know why I didn't think of `when` in any case. I'll change it just for tidiness.